### PR TITLE
Add VIA support for kbd8x mk2

### PR DESCRIPTION
**Pull request by asymy**
## Description

New keymap for via support. The vendor and product ID have already been set from ai03. Added four layers into the keycap (which is otherwise identical to default keymap). 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/kbdfans/kbd8x_mk2/keymaps/via/keymap.c [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+	KC_TRNS, KC_TRNS,  KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,                      KC_TRNS,                                      \
+	KC_TRNS, KC_TRNS,  KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,             KC_TRNS, KC_TRNS,                   KC_TRNS,           \
+	KC_TRNS, KC_TRNS,  KC_TRNS,                                        KC_TRNS,                                                    KC_TRNS,   KC_TRNS,  KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS   \
+  )
+
+
+  
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
```
Comment (mechmerlin) [inline keyboards/kbdfans/kbd8x_mk2/keymaps/via/keymap.c]: Lines 63 to 77 are unneeded

Comment (asymy) [reply keyboards/kbdfans/kbd8x_mk2/keymaps/via/keymap.c]: removed

File: keyboards/kbdfans/kbd8x_mk2/keymaps/via/rules.mk [hunk 2]
```diff
@@ -0,0 +1 @@
+VIA_ENABLE = yes
```
Comment (noroadsleft) [inline keyboards/kbdfans/kbd8x_mk2/keymaps/via/rules.mk]: This compiles fairly close to the chip's capacity; let's enable Link Time Optimization.

```suggestion
VIA_ENABLE = yes
LTO_ENABLE = yes
```

File: keyboards/kbdfans/kbd8x_mk2/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [0] = LAYOUT( /* Layer Zero */
+    
+	KC_ESC,           KC_F1,   KC_F2,   KC_F3,   KC_F4,        KC_F5,   KC_F6,   KC_F7,   KC_F8,       KC_F9,   KC_F10,  KC_F11,  KC_F12,           KC_PSCR, KC_SLCK, KC_PAUS, \
+	KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_DEL,           KC_INS,  KC_HOME, KC_PGUP, \
+	KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC,          KC_BSLS,          KC_DEL,  KC_END,  KC_PGDN, \
+	KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,                   KC_ENT,                                      \
+	KC_LSFT, KC_BSLS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,          KC_RSFT, KC_DEL,                    KC_UP,            \
+	KC_LCTL, KC_LGUI, KC_LALT,                           KC_SPC,                                       KC_RALT, KC_RGUI, KC_APP,  KC_RCTL,          KC_LEFT, KC_DOWN, KC_RGHT  \
```
Comment (fauxpark) [inline keyboards/kbdfans/kbd8x_mk2/keymaps/via/keymap.c]: The backslashes at the end of these lines are not necessary. You can keep them if you want as it isn't hurting anything, but IMO it looks neater without them.

Comment (asymy) [reply keyboards/kbdfans/kbd8x_mk2/keymaps/via/keymap.c]: thanks for the tip, removed as I agree it looks better

**Comment by fauxpark** [comment]
Thanks!

**Pull request merged**